pysteps/blending/linear_blending.py

```diff
@@ -173,6 +177,17 @@ def forecast(
             precip_nwp.shape, precip_nowcast.shape
         )
 
+        # Ensure we are not working with nans in the bleding.
+        # Check if the NWP data contains any nans. If so, fill them with 0.0.
+        precip_nwp = np.nan_to_num(precip_nwp, nan=0.0)
+
+        # Fill nans in precip_nowcast
+        nan_indices = np.isnan(precip_nowcast)
+        if fill_nwp:
+            precip_nowcast[nan_indices] = precip_nwp[nan_indices]
+        else:
+            precip_nowcast[nan_indices] = 0.0
```

Comment:
```suggestion
        nan_mask = np.isnan(precip_nowcast)
        if fill_nwp:
            precip_nowcast[nan_mask] = precip_nwp[nan_mask]
        else:
            precip_nowcast[nan_mask] = 0.0
```
Nitpicky, sorry. But `np.isnan` returns a mask not a list of indices.

Reply:
Good suggestion, thanks!